### PR TITLE
[YUNIKORN-1056] Document usage of scheduler plugin

### PR DESCRIPTION
**Pull request by craigcondit**
https://issues.apache.org/jira/browse/YUNIKORN-1056

### CODE REVIEW

File: docs/developer_guide/deployment.md [hunk 1]
```diff
@@ -98,6 +98,11 @@ The deployment will run 2 containers from your pre-built docker images in 1 pod,
 * yunikorn-scheduler-core (yunikorn scheduler core and shim for K8s)
 * yunikorn-scheduler-web (web UI)
 
+Alternatively, the scheduler can be deployed as a K8S scheduler plugin:
+```
+kubectl create -f deployments/scheduler/plugin.yaml
+```
+
```
Comment (yangwwei) [inline docs/developer_guide/deployment.md:105]: ## Deploy the Scheduler

There are 2 deployment modes for yunikorn, standalone and scheduler-plugin. In standalone mode, yunikorn runs as a customized K8s scheduler and is supposed to replace the default scheduler in the cluster; in the scheduler-plugin mode, yunikorn is built based on K8s scheduling framework as a scheduler-plugin, preserves all default scheduler functionalities plus the yunikorn features. 

### Standalone
// doc to deploy as the standalone scheduler

### scheduler-plugin
// doc to deploy as a plugin

**Review by yangwwei — CHANGES_REQUESTED** [review]
hi @craigcondit thanks for adding this. This is for the developer. Can we add some doc in the user guide as well? Essentially we should have a page to explain what's are the 2 different modes, standalone vs plugin. And what does that mean to the user? I do not think we have anywhere explained this in detail today, which is a major thing missing.

File: docs/developer_guide/deployment.md (hunk 1 above)
Comment (craigcondit) [reply docs/developer_guide/deployment.md:105]: @yangwwei , that's in here as well under the getting started page.

Comment (craigcondit) [reply docs/developer_guide/deployment.md:105]: Added a new comparision page that contrasts the deployment modes.

Comment (yangwwei) [reply docs/developer_guide/deployment.md:105]: Thanks

**Review by yangwwei — APPROVED** [review]
LGTM

**Pull request merged**